functions/save_hours_test.ts

```diff
@@ -1,45 +1,59 @@
-import * as mf from "mock-fetch/mod.ts";
+import { stub } from "@std/testing/mock";
 import { SlackFunctionTester } from "deno-slack-sdk/mod.ts";
-import { assertEquals } from "https://deno.land/std@0.153.0/testing/asserts.ts";
+import { assertEquals } from "@std/assert";
 import SaveHoursFunction from "./save_hours.ts";
 
-// Replaces globalThis.fetch with the mocked copy
-mf.install();
-
-mf.mock("POST@/api/users.profile.get", () => {
-  return new Response(JSON.stringify({
-    ok: true,
-    profile: {
-      real_name: "Cactus Poke",
+function stubFetch() {
+  // Replaces globalThis.fetch with the mocked copy
+  return stub(
+    globalThis,
+    "fetch",
+    async (url: string | URL | Request, options?: RequestInit) => {
+      const req = url instanceof Request ? url : new Request(url, options);
+
+      assertEquals(req.method, "POST");
+
+      switch (req.url) {
+        case "https://slack.com/api/users.profile.get":
+          return new Response(JSON.stringify({
+            ok: true,
+            profile: {
+              real_name: "Cactus Poke",
+            },
+          }));
+        case "https://slack.com/api/apps.auth.external.get": {
+          const body = await req.formData();
+          if (body.get("external_token_id") === "INVALID_TOKEN_ID") {
+            return new Response(JSON.stringify({
+              ok: false,
+              error: "Invalid token",
+            }));
+          }
+          return new Response(JSON.stringify({
+            ok: true,
+            external_token: "GOOGLE_ACCESS_TOKEN",
+          }));
+        }
+        case "https://sheets.googleapis.com/v4/spreadsheets/undefined/values/A2:F2:append?valueInputOption=USER_ENTERED":
+          // Code to execute if expression === value2
+          return new Response(JSON.stringify({
+            ok: true,
+          }));
+        default:
+          throw Error(
+            `No stub found for ${req.method} ${req.url}\nHeaders: ${
+              JSON.stringify(Object.fromEntries(req.headers.entries()))
+            }`,
+          );
+      }
     },
-  }));
-});
-
-mf.mock("POST@/api/apps.auth.external.get", async (args) => {
-  const body = await args.formData();
-
-  if (body.get("external_token_id") === "INVALID_TOKEN_ID") {
-    return new Response(JSON.stringify({
-      ok: false,
-      error: "Invalid token",
-    }));
-  }
-
-  return new Response(JSON.stringify({
-    ok: true,
-    external_token: "GOOGLE_ACCESS_TOKEN",
-  }));
-});
-
-mf.mock("POST@/v4/spreadsheets/*/values/*:append", () => {
-  return new Response(JSON.stringify({
-    ok: true,
-  }));
-});
+  );
+}
 
 const { createContext } = SlackFunctionTester("save_hours");
 
 Deno.test("Fail on invalid auth token id", async () => {
+  using _stubFetch = stubFetch();
```

Comment:
🧠 TIL that `using` scopes these **global** stubs to the function scope! How interesting!